drivers/sensor/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,198 @@
+
+#ifdef CONFIG_DS18B20_RESOLUTION_9BIT
+	#define DS18B20_RESOLUTION_BITS 0x00
+	#define DS18B20_MAX_CONVERSION_TIME 93750
+#endif
+
+#ifdef CONFIG_DS18B20_RESOLUTION_10BIT
+	#define DS18B20_RESOLUTION_BITS (1 << DS18B20_CONFIG_BIT_R0)
+	#define DS18B20_MAX_CONVERSION_TIME 187500
+#endif
+
+#ifdef CONFIG_DS18B20_RESOLUTION_11BIT
+	#define DS18B20_RESOLUTION_BITS (1 << DS18B20_CONFIG_BIT_R1)
+	#define DS18B20_MAX_CONVERSION_TIME 375000
+#endif
+
+#ifdef CONFIG_DS18B20_RESOLUTION_12BIT
+	#define DS18B20_RESOLUTION_BITS \
+		((1 << DS18B20_CONFIG_BIT_R0) | (1 << DS18B20_CONFIG_BIT_R1))
+	#define DS18B20_MAX_CONVERSION_TIME 750000
+#endif
+
+#define DS18B20_T_HIGH(t) ((t & 0x0FFF) >> 4)
+#define DS18B20_T_LOW(t) ((t & 0x000F) * 10000)
+
+static int ds18b20_write_scratchpad(struct device *dev,
+	ds18b20_scratchpad_t scratchpad)
+{
+
+	struct ds18b20_data *drv_data = dev->driver_data;
+	struct device *bus = drv_data->bus;
+
+	u8_t data[3] = {
+		scratchpad.alarm_temperature_high,
+		scratchpad.alarm_temperature_low,
+		scratchpad.configuration
+	};
+
+	w1_skip_rom(bus);
+	w1_send_command(bus, DS18B20_COMMAND_WRITE_SCRATCHPAD);
+	w1_write_block(bus, &data, sizeof(data));
+
+	return 0;
+}
+
+static int ds18b20_read_scratchpad(struct device *dev,
+	ds18b20_scratchpad_t *scratchpad)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+	struct device *bus = drv_data->bus;
+
+	w1_skip_rom(bus);
+	w1_send_command(bus, DS18B20_COMMAND_READ_SCRATCHPAD);
+	w1_read_block(bus, (char *)scratchpad, 9);
+
+	return 0;
+}
+
+static void ds18b20_convert(struct device *dev)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+	struct device *bus = drv_data->bus;
+
+	w1_skip_rom(bus);
+	w1_send_command(drv_data->bus, DS18B20_COMMAND_CONVERT_T);
+	w1_wait_for(drv_data->bus, 1, DS18B20_MAX_CONVERSION_TIME);
+}
+
+static int ds18b20_read_power_source(struct device *dev)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+	struct device *bus = drv_data->bus;
+
+	w1_skip_rom(bus);
+	w1_send_command(bus, DS18B20_COMMAND_READ_POWER_SOURCE);
+	drv_data->power_source = w1_read_bit(bus);
+
+	return 0;
+}
+
+static int ds18b20_set_resolution(struct device *dev)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+
+	drv_data->scratchpad.configuration &= \
+		~((1 << DS18B20_CONFIG_BIT_R0) | (1 << DS18B20_CONFIG_BIT_R1));
+	drv_data->scratchpad.configuration |= DS18B20_RESOLUTION_BITS;
+
+	return 0;
+}
+
+static int ds18b20_set_alarm_trigger(struct device *dev, u8_t low, u8_t high)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+
+	drv_data->scratchpad.alarm_temperature_high = high;
+	drv_data->scratchpad.alarm_temperature_low = low;
+
+	return 0;
+}
+
+static int ds18b20_channel_get(struct device *dev,
+			enum sensor_channel chan,
+			struct sensor_value *val)
+{
+	struct ds18b20_data *drv_data = dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_TEMP);
+
+	ds18b20_read_scratchpad(dev, &drv_data->scratchpad);
+
+	if (chan == SENSOR_CHAN_TEMP) {
```

Comment:
shouldn't this be `SENSOR_CHAN_AMBIENT_TEMP`?

Reply:
Fixed